eventuals/do-all.h

```diff
@@ -93,64 +106,65 @@ struct _DoAll final {
                    }
                    if (counter_.fetch_sub(1) == 1) {
                      // You're the last eventual so call the continuation.
-                     std::optional<std::exception_ptr> exception =
-                         GetExceptionIfExists();
-
-                     if (exception) {
-                       try {
-                         std::rethrow_exception(*exception);
-                       } catch (const StoppedException&) {
-                         k_.Stop();
-                       } catch (...) {
-                         k_.Fail(std::current_exception());
-                       }
+                     std::optional<StoppedOrError> stopped_or_error =
+                         GetStoppedOrErrorIfExists();
+
+                     if (stopped_or_error) {
+                       std::visit(
+                           [this](auto&& error) {
```

Comment:
```suggestion
                           [this](auto&& stopped_or_error) {
```
And this is consistent with what you had in a previous file I took a look at too!